src/handlers/web3.ts

```diff
@@ -0,0 +1,683 @@
+    const { gasLimit } = await (p.getBlock as () => Promise<Block>)();
+
+    const { to, data } = txPayloadToEstimate;
+
+    // 1 - Check if the receiver is a contract
+    const code = to ? await p.getCode(to) : undefined;
+    // 2 - if it's not a contract AND it doesn't have any data use the default gas limit
+    if (
+      (!contractCallEstimateGas && !to) ||
+      (to && !data && (!code || code === '0x'))
+    ) {
+      logger.sentry(
+        '⛽ Skipping estimates, using default',
+        ethUnits.basic_tx.toString()
+      );
+      return ethUnits.basic_tx.toString();
+    }
+
+    logger.sentry('⛽ Calculating safer gas limit for last block');
+    // 3 - If it is a contract, call the RPC method `estimateGas` with a safe value
+    const saferGasLimit = fraction(gasLimit.toString(), 19, 20);
+    logger.sentry('⛽ safer gas limit for last block is', saferGasLimit);
+
+    txPayloadToEstimate[contractCallEstimateGas ? 'gasLimit' : 'gas'] = toHex(
+      saferGasLimit
+    );
+
+    const estimatedGas = await (contractCallEstimateGas
+      ? contractCallEstimateGas(...(callArguments ?? []), txPayloadToEstimate)
+      : p.estimateGas(txPayloadToEstimate));
+
+    const lastBlockGasLimit = addBuffer(gasLimit.toString(), 0.9);
+    const paddedGas = addBuffer(
+      estimatedGas.toString(),
+      paddingFactor.toString()
+    );
+    logger.sentry('⛽ GAS CALCULATIONS!', {
+      estimatedGas: estimatedGas.toString(),
+      gasLimit: gasLimit.toString(),
+      lastBlockGasLimit: lastBlockGasLimit,
+      paddedGas: paddedGas,
+    });
+
+    // If the safe estimation is above the last block gas limit, use it
+    if (greaterThan(estimatedGas.toString(), lastBlockGasLimit)) {
+      logger.sentry(
+        '⛽ returning orginal gas estimation',
+        estimatedGas.toString()
+      );
+      return estimatedGas.toString();
+    }
+    // If the estimation is below the last block gas limit, use the padded estimate
+    if (greaterThan(lastBlockGasLimit, paddedGas)) {
+      logger.sentry('⛽ returning padded gas estimation', paddedGas);
+      return paddedGas;
+    }
+    // otherwise default to the last block gas limit
+    logger.sentry('⛽ returning last block gas limit', lastBlockGasLimit);
+    return lastBlockGasLimit;
+  } catch (error) {
+    logger.error('Error calculating gas limit with padding', error);
+    return null;
+  }
+}
+
+/**
+ * @desc convert from ether to wei
+ * @param value The value in ether.
+ * @return The value in wei.
+ */
+export const toWei = (ether: string): string => {
+  const result = parseEther(ether);
+  return result.toString();
+};
+
+/**
+ * @desc get transaction info
+ * @param hash The transaction hash.
+ * @return The corresponding `TransactionResponse`, or `null` if one could not
+ * be found.
+ */
+export const getTransaction = async (
+  hash: string
+): Promise<TransactionResponse | null> =>
+  web3Provider ? web3Provider.getTransaction(hash) : null;
```

Comment:
```suggestion
  web3Provider?.getTransaction(hash) ?? null;
```

or even

```suggestion
  web3Provider?.getTransaction(hash);
```
+ change the return type